### ISSUE TITLE
Warning for `Trainer` with `online_measurements` and `physics_generator`

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#202]
When using `Trainer` with `online_measurements=True` and `physics_generator is not None`, a new set of physics params is generated per batch, which changes every epoch. We could reset the `physics_generator` every epoch, but the problem will still persist if the dataloader has `shuffle=True`. Some users may assume that the generated online measurements are same every epoch but this is not so. For some, this may be desired behaviour, but it is not always so. I think we should warn users of this potentially unexpected behaviour.

Thanks!

**Comment by tachella** [comment]
Hi Andrew,

I think it is expected to have different params at every training step. If a user wants the same params in all training steps, it would suffice to either
1. set them in the physics operator before training, or
2. create a physics generator with a deterministic generation (basically giving the same thing each time)

**Comment by tachella** [comment]
If I'm not missing something, setting them in the physics operator before training would be enough?
nonetheless, I think for this setting, it is more clean to use multiple physics operators.

**Comment by Andrewwango** [comment]
Yes, multiple physics operators also works. I still think it's worth adding a warning though as the behaviour might not be intended if you are not expecting the set of generator params to change every epoch.
